CompilerSource/compiler/components/parse_and_link.cpp

```diff
@@ -95,7 +95,7 @@ int lang_CPP::compile_parseAndLink(EnigmaStruct *es,parsed_script *scripts[], ve
       // Keep a parsed record of this timeline
       tline_lookup[es->timelines[i].name].push_back(tlines.back());
       parser_main(newcode, &tlines.back()->pev, script_names);
-      edbg << "Parsed `" << es->timelines[i].name <<", moment: " <<es->timelines[i].moments[j].stepNo << "': " << tlines.back()->obj.locals.size() << " locals, " << tlines.back()->obj.globals.size() << " globals" << flushl;
+      edbg << "Parsed `" << es->timelines[i].name <<", moment: " <<es->timelines[i].moments[j].stepNo << "': " << (int)(tlines.back()->obj.locals.size()) << " locals, " << (int)tlines.back()->obj.globals.size() << " globals" << flushl;
```

Comment:
When I remove these casts I get these errors:

```
compiler/components/parse_and_link.cpp:98:112: error: use of overloaded operator '<<' is
      ambiguous (with operand types 'ideprint' and 'size_type' (aka 'unsigned long'))
  ...edbg << "Parsed `" << es->timelines[i].name <<", moment: " <<es->timelines[i].moments[j].stepNo << "': " << (tlines.back()->obj.locals.size())...
     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ ^  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./backend/ideprint.h:29:13: note: candidate function
  ideprint &operator<< (int16_t x);
            ^
./backend/ideprint.h:30:13: note: candidate function
  ideprint &operator<< (int32_t x);
            ^
./backend/ideprint.h:31:13: note: candidate function
  ideprint &operator<< (int64_t x);
            ^
./backend/ideprint.h:32:13: note: candidate function
  ideprint &operator<< (uint16_t x);
            ^
./backend/ideprint.h:33:13: note: candidate function
  ideprint &operator<< (uint32_t x);
            ^
./backend/ideprint.h:34:13: note: candidate function
  ideprint &operator<< (uint64_t x);
            ^
./backend/ideprint.h:35:13: note: candidate function
  ideprint &operator<< (char x);
            ^
./backend/ideprint.h:36:13: note: candidate function
  ideprint &operator<< (unsigned char x);
            ^
./backend/ideprint.h:37:13: note: candidate function
  ideprint &operator<< (double x);
            ^
```
_(And the same error for macros.cpp)_

---
So, I tried to properly fix it without using these casts by adding a
`ideprint &operator<< (size_t x);`

**It fixed the parse_and_link file without the use of casts** :)

But not the macros.cpp :(
I got some linker errors:
```
Undefined symbols for architecture x86_64:
  "parse_bacics::visible::toString(unsigned long)", referenced from:
      jdip::macro_function::parse(std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > >&, char*&, char*&, jdip::token_t, jdi::error_handler*) const in macros.o
      jdip::macro_function::preparse(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, jdi::error_handler*) in macros.o
ld: symbol(s) not found for architecture x86_64
```

Looks like a template problem.

Not sure how to fix, any tips?